samples/bluetooth/per_sync/src/main.c

```diff
@@ -0,0 +1,287 @@
+	       bt_le_per_adv_sync_get_index(sync), le_addr, info->tx_power,
+	       info->rssi, info->cte_type, buf->len);
+}
+
+static struct bt_le_per_adv_sync_cb sync_callbacks = {
+	.synced = sync_cb,
+	.term = term_cb,
+	.recv = recv_cb
+};
+
+void main(void)
+{
+	struct bt_le_per_adv_sync_param sync_create_param;
+	struct bt_le_per_adv_sync *sync;
+	int err;
+
+	printk("Starting Periodic Advertising Synchronization Demo\n");
+
+#if defined(HAS_LED)
+	printk("Get reference to LED device...");
+	dev = device_get_binding(LED0);
+	if (!dev) {
+		printk("Failed.\n");
+		return;
+	}
+	printk("done.\n");
+
+	printk("Configure GPIO pin...");
+	err = gpio_pin_configure(dev, PIN, GPIO_OUTPUT_ACTIVE | FLAGS);
+	if (err) {
+		return;
+	}
+	printk("done.\n");
+
+	k_delayed_work_init(&blink_work, blink_timeout);
+#endif /* HAS_LED */
+
+	/* Initialize the Bluetooth Subsystem */
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Scan callbacks register...");
+	bt_le_scan_cb_register(&scan_callbacks);
+	printk("success.\n");
+
+	printk("Periodic Advertising callbacks register...");
+	bt_le_per_adv_sync_cb_register(&sync_callbacks);
+	printk("Success.\n");
+
+	printk("Initialize semaphore sem_per_adv...");
+	err = k_sem_init(&sem_per_adv, 0, 1);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success.\n");
+
+	printk("Initialize semaphore sem_per_sync...");
+	err = k_sem_init(&sem_per_sync, 0, 1);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success.\n");
+
+	printk("Initialize semaphore sem_per_sync_lost...");
+	err = k_sem_init(&sem_per_sync_lost, 0, 1);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success.\n");
+
+	printk("Start scanning...");
+	err = bt_le_scan_start(BT_LE_SCAN_ACTIVE, NULL);
+	if (err) {
+		printk("failed (err %d)\n", err);
+		return;
+	}
+	printk("success.\n");
+
+	do {
+#if defined(HAS_LED)
+		printk("Start blinking LED...\n");
+		led_is_on = false;
+		gpio_pin_set(dev, PIN, (int)led_is_on);
+		k_delayed_work_submit(&blink_work, BLINK_ONOFF);
+#endif /* HAS_LED */
+
+		printk("Waiting for periodic advertising...\n");
+		is_per_adv = false;
+		err = k_sem_take(&sem_per_adv, K_FOREVER);
+		if (err) {
+			printk("failed (err %d)\n", err);
+			return;
+		}
+		printk("Found periodic advertising.\n");
+
+		printk("Creating Periodic Advertising Sync...");
+		bt_addr_le_copy(&sync_create_param.addr, &per_addr);
+		sync_create_param.options = 0;
+		sync_create_param.sid = per_sid;
+		sync_create_param.skip = 0;
+		sync_create_param.timeout = 0xa;
+		err = bt_le_per_adv_sync_create(&sync_create_param, &sync);
+		if (err) {
+			printk("failed (err %d)\n", err);
+			return;
+		}
+		printk("success.\n");
+
+		printk("Waiting for periodic sync...\n");
+		err = k_sem_take(&sem_per_sync, K_FOREVER);
+		if (err) {
+			printk("failed (err %d)\n", err);
+			continue;
```

Comment:
I assume that we in all cases should return from main when we fail to take a semaphore with infinite timeout?
```suggestion
			return;
```

Reply:
action: I will use a timeout and cancel sync create. It will be difficult to test (to kill periodic advertiser before the scanner sync after it has found it).

Reply:
done